fineract-loan/src/main/java/org/apache/fineract/accounting/producttoaccountmapping/service/ProductToGLAccountMappingWritePlatformServiceImpl.java

```diff
@@ -278,6 +278,77 @@ public void createSavingProductToGLAccountMapping(final Long savingProductId, fi
                 this.savingsProductToGLAccountMappingHelper.savePaymentChannelToFundSourceMappings(command, element, savingProductId, null);
                 this.savingsProductToGLAccountMappingHelper.saveChargesToIncomeAccountMappings(command, element, savingProductId, null);
             break;
+
+            case ACCRUAL_PERIODIC:
+                // asset
```

Comment:
It would be better to extract these into a method and then you can extract the common mappings of cash and accrual based mapping also to reduce code duplications.